httpx/_auth.py

```diff
@@ -46,6 +51,56 @@ def auth_flow(self, request: Request) -> typing.Generator[Request, Response, Non
         """
```

Comment:
Should we raise a `NotImplementedError()` here by default?

Reply:
Okay, I can see we're using `Auth()` in no-auth cases at the moment, so no that wouldn't work trivially.
Might be something to think about there, but let's treat it separately if we do.